app/controllers/admin/collections_controller.rb

```diff
@@ -217,4 +217,40 @@ def destroy
       redirect_to admin_collection_path(@source_collection)
     end
   end
+
+  # POST /collections/1/batch_ingest
+  def batch_ingest
+    if @collection.dropbox.manifest_processable?
+      # Spawn off background job
+      AvalonJobs.collection_batch_ingest(@collection.id)
+      flash[:notice] = t('batch_ingest.initiated', name: @collection.name)
+    else
+      flash[:error] =
+        if @collection.dropbox.manifest_error?
```

Comment:
I would move all this logic down into the dropbox class?

Could be just...

`flash[:error] = @collection.dropbox.manifest_status_message`

Or even better....just intepolate the status into the i18n key

`flash[:error] =  t("batch_ingest.#{@collection.dropbox.manifest_status}", name: @collection.name)`

no more crazy branching logic!